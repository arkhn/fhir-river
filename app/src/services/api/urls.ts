let API_URL = "" as string | undefined;
if (process.env.NODE_ENV === "production") {
  ({ REACT_APP_API_URL: API_URL } = (window as any).env);
} else {
  ({ REACT_APP_API_URL: API_URL } = process.env);
}

export { API_URL };
