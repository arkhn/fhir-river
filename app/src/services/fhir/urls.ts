export const { REACT_APP_FHIR_API_URL: FHIR_API_URL } = process.env;
