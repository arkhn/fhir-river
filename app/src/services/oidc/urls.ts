export const {
  REACT_APP_OIDC_LOGIN_URL: OIDC_LOGIN_URL,
  REACT_APP_OIDC_LOGOUT_URL: OIDC_LOGOUT_URL,
} = process.env;
