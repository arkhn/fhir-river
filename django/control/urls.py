from django.urls import include, path

from control.api import views
from control.api.router import router

urlpatterns = [
    path("", include(router.urls)),
    path("preview/", views.PreviewEndpoint.as_view(), name="preview"),
    path("delete-resources/", views.ResourceEndpoint.as_view(), name="delete-resources"),
]
