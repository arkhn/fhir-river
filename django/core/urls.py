from django.urls import include, path

from core.api import views
from core.api.router import router

urlpatterns = [
    path("", include(router.urls)),
    path("version/", views.VersionEndpoint.as_view()),
]
