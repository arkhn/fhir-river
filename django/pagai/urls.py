from django.urls import path

from pagai.api import views

urlpatterns = [
    path("pagai/list-owners/", views.OwnersListView.as_view(), name="list-owners"),
    path("pagai/owner-schema/<owner>/", views.OwnerSchemaView.as_view(), name="owner-schema"),
    path("pagai/explore/<resource_id>/<owner>/<table>/", views.ExploreView.as_view(), name="explore"),
]
