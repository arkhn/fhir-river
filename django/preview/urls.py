from django.urls import include, path

from preview.api.router import router
from preview.api import views

urlpatterns = [
    path("", include(router.urls)),
    path("preview/", views.PreviewEndpoint.as_view(), name="preview"),
]
