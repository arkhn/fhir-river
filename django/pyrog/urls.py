from django.urls import include, path

from pyrog.api import views
from pyrog.api.router import router


urlpatterns = [
    path("api/", include(router.urls)),
    path('session/', views.Session.as_view()),
    path('oidc/', include('mozilla_django_oidc.urls')),
]
