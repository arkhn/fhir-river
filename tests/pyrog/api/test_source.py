import pytest
from faker import Faker

from django.urls import reverse

from dateutil.parser import parse

faker = Faker()

pytestmark = pytest.mark.django_db


@pytest.mark.parametrize("name, version, status_code", [(faker.word(), faker.word(), 201), ("", faker.word(), 400)])
def test_create_source(api_client, name, version, status_code):
    url = reverse("sources-list")

    data = {"name": name, "version": version}
    response = api_client.post(url, data)

    assert response.status_code == status_code, response.data


@pytest.mark.as_user
@pytest.mark.parametrize("name, version, status_code", [(faker.word(), faker.word(), 201), ("", faker.word(), 400)])
def test_create_and_assign_source(api_client, user, name, version, status_code):
    url = reverse("sources-list")

    data = {"name": name, "version": version}
    response = api_client.post(url, data)

    assert response.status_code == status_code, response.data

    if status_code != 201:
        return

    # Check that the new source has been assigned to the authenticated user
    assert user.sources.filter(id=response.data["id"]).exists()


@pytest.mark.as_user
def test_retrieve_source(api_client, source):
    url = reverse("sources-detail", kwargs={"pk": source.id})

    response = api_client.get(url)

    assert response.status_code == 200, response.data


def test_retrieve_source_unauthenticated(api_client, source):
    url = reverse("sources-detail", kwargs={"pk": source.id})

    response = api_client.get(url)

    assert response.status_code == 404, response.data


@pytest.mark.as_other_user
def test_retrieve_source_forbidden(api_client, source):
    url = reverse("sources-detail", kwargs={"pk": source.id})

    response = api_client.get(url)

    assert response.status_code == 404, response.data


@pytest.mark.as_user
@pytest.mark.export_data("valid/0003.json")
def test_can_import_mapping(api_client, export_data):
    url = reverse("sources-list")

    response = api_client.post(url + "import/", export_data, format="json")

    assert response.status_code == 201, response.data


@pytest.mark.as_user
@pytest.mark.export_data("valid/0003.json")
def test_can_import_mapping_several_times(api_client, export_data):
    url = reverse("sources-list")

    response = api_client.post(url + "import/", export_data, format="json")
    assert response.status_code == 201, response.data

    # Change source name
    export_data["name"] = "other_name"
    response = api_client.post(url + "import/", export_data, format="json")
    assert response.status_code == 201, response.data

    response = api_client.get(url)
    assert len(response.data) == 2


@pytest.mark.as_user
def test_list_sources(api_client, user, other_user, source_factory):
    url = reverse("sources-list")

    # Assign 3 sources to the authenticated user
    source_factory.create_batch(3, source_user__user=user)
    assert user.sources.count() == 3
    assert other_user.sources.count() == 0
    # Assign 3 other sources to the other user
    source_factory.create_batch(3, source_user__user=other_user)
    assert user.sources.count() == 3
    assert other_user.sources.count() == 3

    response = api_client.get(url)

    assert response.status_code == 200, response.data
    assert {source.id for source in user.sources.all()} == {item["id"] for item in response.data}
    assert all(source.id not in {item["id"] for item in response.data} for source in other_user.sources.all())
    assert all(
        parse(response.data[i]["created_at"]) <= parse(response.data[i + 1]["created_at"])
        for i in range(len(response.data) - 1)
    )


def test_list_sources_unauthenticated(api_client, source):
    url = reverse("sources-list")

    response = api_client.get(url)

    assert response.status_code == 200, response.data
    assert len(response.data) == 0


@pytest.mark.as_user
@pytest.mark.parametrize("name, version, status_code", [(faker.word(), None, 200), (None, faker.word(), 200)])
def test_update_source(api_client, source, name, version, status_code):
    url = reverse("sources-detail", kwargs={"pk": source.id})

    data = {}
    if name:
        data["name"] = name
    if version:
        data["version"] = version
    response = api_client.patch(url, data)

    assert response.status_code == status_code, response.data


def test_update_source_unauthenticated(api_client, source):
    url = reverse("sources-detail", kwargs={"pk": source.id})

    data = {"name": faker.word()}
    response = api_client.patch(url, data)

    assert response.status_code == 404, response.data


@pytest.mark.as_other_user
def test_update_source_forbidden(api_client, source):
    url = reverse("sources-detail", kwargs={"pk": source.id})

    data = {"name": faker.word()}
    response = api_client.patch(url, data)

    assert response.status_code == 404, response.data


@pytest.mark.as_user
def test_delete_source(api_client, source):
    url = reverse("sources-detail", kwargs={"pk": source.id})

    response = api_client.delete(url)

    assert response.status_code == 204, response.data


def test_delete_source_unauthenticated(api_client, source):
    url = reverse("sources-detail", kwargs={"pk": source.id})

    response = api_client.delete(url)

    assert response.status_code == 404, response.data


@pytest.mark.as_other_user
def test_delete_source_forbidden(api_client, source):
    url = reverse("sources-detail", kwargs={"pk": source.id})

    response = api_client.delete(url)

    assert response.status_code == 404, response.data
