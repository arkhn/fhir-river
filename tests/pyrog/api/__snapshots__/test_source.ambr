# name: test_retrieve_full_source
  <class 'dict'> {
    'credential': <class 'dict'> {
      'database': '',
      'host': '',
      'model': '',
      'owners': <class 'list'> [
        <class 'dict'> {
          'columns': <class 'list'> [
            <class 'dict'> {
              'column': '',
              'id': 'column_id_0001',
              'joins': <class 'list'> [
              ],
              'table': '',
            },
          ],
          'id': 'owner_id_0001',
          'name': 'owner_1',
          'schema': None,
        },
      ],
      'port': 5432,
    },
    'name': 'source_1',
    'resources': <class 'list'> [
      <class 'dict'> {
        'attributes': <class 'list'> [
          <class 'dict'> {
            'definition_id': '',
            'input_groups': <class 'list'> [
              <class 'dict'> {
                'conditions': <class 'list'> [
                  <class 'dict'> {
                    'action': '',
                    'column': 'column_id_0001',
                    'relation': 'EQ',
                    'value': '',
                  },
                ],
                'inputs': <class 'list'> [
                  <class 'dict'> {
                    'column': 'column_id_0001',
                    'concept_map_id': '',
                    'script': '',
                    'static_value': '',
                  },
                ],
                'merging_script': '',
              },
            ],
            'path': '',
            'slice_name': '',
          },
        ],
        'definition_id': '',
        'filters': <class 'list'> [
        ],
        'label': '',
        'logical_reference': '',
        'primary_key_column': '',
        'primary_key_owner': 'owner_id_0001',
        'primary_key_table': '',
      },
    ],
    'version': '',
  }
---
